From b49dce9f3023210cd501bbe11d992ab2ab6d143b Mon Sep 17 00:00:00 2001
From: Jameson Ernst <jameson@jpernst.com>
Date: Mon, 10 Jul 2017 12:50:30 -0700
Subject: [PATCH 12/23] Remove slice on listener orientation to reduce garbage

---
 src/library_openal.js | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/library_openal.js b/src/library_openal.js
index b75dba1a8..c7c8ffaad 100644
--- a/src/library_openal.js
+++ b/src/library_openal.js
@@ -854,8 +854,12 @@ var LibraryOpenAL = {
           return;
         }
 
-        listener._direction = value.slice(0, 3);
-        listener._up = value.slice(3, 6);
+        listener._direction[0] = value[0];
+        listener._direction[1] = value[1];
+        listener._direction[2] = value[2];
+        listener._up[0] = value[3];
+        listener._up[1] = value[4];
+        listener._up[2] = value[5];
         AL.updateListenerSpace(AL.currentCtx);
         break;
       default:
-- 
2.14.1

