From a588e771067a8833b1dffe9359fc2d20f52877af Mon Sep 17 00:00:00 2001
From: yoanlcq <yoanlecoq.io@gmail.com>
Date: Mon, 10 Jul 2017 23:45:03 +0200
Subject: [PATCH 15/23] capture: Fix regression in onaudioprocess from previous
 'optimization'

---
 src/library_openal.js | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/library_openal.js b/src/library_openal.js
index ffcd80f61..4044e844f 100644
--- a/src/library_openal.js
+++ b/src/library_openal.js
@@ -1771,7 +1771,7 @@ var LibraryOpenAL = {
           var channel0 = srcBuf.getChannelData(0);
           for (var i = 0 ; i < srcBuf.length; ++i) {
             var wi = (c.capturePlayhead + i) % c.bufferFrameCapacity;
-            c.buffers[0][wi] = channel0[i] * 0.000030517578125 /* 1/32768 */;
+            c.buffers[0][wi] = channel0[i] * 32767;
           }
           break;
         case 0x1103:  /* AL_FORMAT_STEREO16 */
@@ -1779,15 +1779,15 @@ var LibraryOpenAL = {
           var channel1 = srcBuf.getChannelData(1);
           for (var i = 0 ; i < srcBuf.length; ++i) {
             var wi = (c.capturePlayhead + i) % c.bufferFrameCapacity;
-            c.buffers[0][wi] = channel0[i] * 0.000030517578125 /* 1/32768 */;
-            c.buffers[1][wi] = channel1[i] * 0.000030517578125 /* 1/32768 */;
+            c.buffers[0][wi] = channel0[i] * 32767;
+            c.buffers[1][wi] = channel1[i] * 32767;
           }
           break;
         case 0x1100:  /* AL_FORMAT_MONO8 */
           var channel0 = srcBuf.getChannelData(0);
           for (var i = 0 ; i < srcBuf.length; ++i) {
             var wi = (c.capturePlayhead + i) % c.bufferFrameCapacity;
-            c.buffers[0][wi] = (channel0[i] + 1.0) * 0.0078125 /* 1/128 */;
+            c.buffers[0][wi] = (channel0[i] + 1.0) * 127;
           }
           break;
         case 0x1102:  /* AL_FORMAT_STEREO8 */
@@ -1795,8 +1795,8 @@ var LibraryOpenAL = {
           var channel1 = srcBuf.getChannelData(1);
           for (var i = 0 ; i < srcBuf.length; ++i) {
             var wi = (c.capturePlayhead + i) % c.bufferFrameCapacity;
-            c.buffers[0][wi] = (channel0[i] + 1.0) * 0.0078125 /* 1/128 */;
-            c.buffers[1][wi] = (channel1[i] + 1.0) * 0.0078125 /* 1/128 */;
+            c.buffers[0][wi] = (channel0[i] + 1.0) * 127;
+            c.buffers[1][wi] = (channel1[i] + 1.0) * 127;
           }
           break;
         }
-- 
2.14.1

