From 6160775a96db3c97c3d2f0d0ce506690446e60ed Mon Sep 17 00:00:00 2001
From: Jameson Ernst <jameson@jpernst.com>
Date: Mon, 10 Jul 2017 22:05:24 -0700
Subject: [PATCH 16/23] HRTF panning model should be uppercase

---
 src/library_openal.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/library_openal.js b/src/library_openal.js
index 4044e844f..afb965a94 100644
--- a/src/library_openal.js
+++ b/src/library_openal.js
@@ -426,7 +426,7 @@ var LibraryOpenAL = {
       panner.maxDistance = src.maxDistance;
       panner.rolloffFactor = src.rolloffFactor;
 
-      panner.panningModel = src.context.hrtf ? "hrtf" : "equalpower";
+      panner.panningModel = src.context.hrtf ? "HRTF" : "equalpower";
 
       // Use the source's distance model if AL_SOURCE_DISTANCE_MODEL is enabled
       var distanceModel = src.context.sourceDistanceModel ? src.distanceModel : src.context.distanceModel;
-- 
2.14.1

