From 8dc77a959ad401973f38e86b9755a8bd11ad00bd Mon Sep 17 00:00:00 2001
From: Jameson Ernst <jameson@jpernst.com>
Date: Sat, 19 Aug 2017 13:12:23 -0700
Subject: [PATCH 19/23] Fix possibility of non-orthonormal basis in source
 relative transform

---
 src/library_openal.js | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/library_openal.js b/src/library_openal.js
index 48dd3d8b2..4a51c74cf 100644
--- a/src/library_openal.js
+++ b/src/library_openal.js
@@ -547,6 +547,17 @@ var LibraryOpenAL = {
         var lRightY = (lUpZ * lBackX - lUpX * lBackZ);
         var lRightZ = (lUpX * lBackY - lUpY * lBackX);
 
+        // Back and Up might not be exactly perpendicular, so the cross product also needs normalization
+        var invMag = 1.0 / Math.sqrt(lRightX * lRightX + lRightY * lRightY + lRightZ * lRightZ);
+        lRightX *= invMag;
+        lRightY *= invMag;
+        lRightZ *= invMag;
+
+        // Recompute Up from the now orthonormal Right and Back vectors so we have a fully orthonormal basis
+        var lUpX = (lBackY * lRightZ - lBackZ * lRightY);
+        var lUpY = (lBackZ * lRightX - lBackX * lRightZ);
+        var lUpZ = (lBackX * lRightY - lBackY * lRightX);
+
         var oldX = dirX;
         var oldY = dirY;
         var oldZ = dirZ;
-- 
2.14.1

