From f5fbea752ebc7a33392d768d941e5b7d2778b974 Mon Sep 17 00:00:00 2001
From: Jameson Ernst <jameson@jpernst.com>
Date: Sat, 19 Aug 2017 13:30:23 -0700
Subject: [PATCH 20/23] Fix source position/orientation compatibility fallbacks

---
 src/library_openal.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/library_openal.js b/src/library_openal.js
index 4a51c74cf..a7420c376 100644
--- a/src/library_openal.js
+++ b/src/library_openal.js
@@ -591,7 +591,7 @@ var LibraryOpenAL = {
 #if OPENAL_DEBUG
         Runtime.warnOnce('Panner position attributes are not present, falling back to setPosition()');
 #endif
-        panner.setPosition(src.position[0], src.position[1], src.position[2]);
+        panner.setPosition(posX, posY, posZ);
       }
       if (panner.orientationX) {
         panner.orientationX.value = dirX;
@@ -601,7 +601,7 @@ var LibraryOpenAL = {
 #if OPENAL_DEBUG
         Runtime.warnOnce('Panner orientation attributes are not present, falling back to setOrientation()');
 #endif
-        panner.setOrientation(val[0], val[1], val[2]);
+        panner.setOrientation(dirX, dirY, dirZ);
       }
 
       var oldShift = src.dopplerShift;
-- 
2.14.1

