From b33fa833010e100a3d532ae4bd4533e494095345 Mon Sep 17 00:00:00 2001
From: Jameson Ernst <jameson@jpernst.com>
Date: Wed, 30 Aug 2017 08:52:50 -0700
Subject: [PATCH 22/23] Fix `AL.buffer` typo

---
 src/library_openal.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/library_openal.js b/src/library_openal.js
index 2d4597e66..35aa05f86 100644
--- a/src/library_openal.js
+++ b/src/library_openal.js
@@ -4065,7 +4065,7 @@ var LibraryOpenAL = {
 
     /// If the queue is empty, put the zero buffer back in
     if (src.bufQueue.length === 0) {
-      src.bufQueue.push(AL.buffer[0]);
+      src.bufQueue.push(AL.buffers[0]);
     }
 
     AL.initSourcePanner(src);
-- 
2.14.1

