From e06b3ac370e84480e7d3c6d9e8184c27fc2d0bf5 Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdisabwat@gmail.com>
Date: Tue, 17 Sep 2019 18:59:43 +0200
Subject: [PATCH 08/15] compat: add sigwait support for emscripten

---
 compat/sigwait.c | 4 ++--
 configure.ac     | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/compat/sigwait.c b/compat/sigwait.c
index e5a082d2d6..950579cc8b 100644
--- a/compat/sigwait.c
+++ b/compat/sigwait.c
@@ -24,8 +24,8 @@
 # include <config.h>
 #endif
 
-#ifdef __native_client__
-/* NaCl has no working sigwait, but SIGPIPE, for which vlc uses sigwait
+#if defined(__native_client__) || defined(__EMSCRIPTEN__) 
+/* NaCl and Emscripten have no working sigwait, but SIGPIPE, for which vlc uses sigwait
  * currently, is never generated in NaCl. So for SIGPIPE it's safe to instantly
  * return, for all others run into an assertion. */
 
diff --git a/configure.ac b/configure.ac
index c9d04253cd..e5298e6f7a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -324,6 +324,7 @@ case "${host_os}" in
     ;;
   *emscripten*)
     SYS=emscripten
+    AC_LIBOBJ([sigwait])
     CC=emcc
     LD=emcc
     LDSHARED=emcc
-- 
2.23.0

