From 9ab65e4debb1f0e5499fcb521e21fa267ff00a55 Mon Sep 17 00:00:00 2001
From: Etienne Brateau <etienne.brateau@gmail.com>
Date: Thu, 6 Jul 2017 14:50:28 +0200
Subject: [PATCH 01/18] wasm-emscripten: Create a target for emscripten in
 configure.ac

---
 configure.ac | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/configure.ac b/configure.ac
index 374ce00cc7..af0a57f774 100644
--- a/configure.ac
+++ b/configure.ac
@@ -378,6 +378,12 @@ case "${host_os}" in
     AC_LIBOBJ([recvmsg])
     AC_LIBOBJ([sendmsg])
     ;;
+  *emscripten*)
+    SYS=emscripten
+    # tdestroy() is a GNU extension
+    CFLAGS="${CFLAGS} -pthread -D_GNU_SOURCE"
+    CXXFLAGS="${CXXFLAGS} -pthread"
+    ;;
   *)
     SYS="${host_os}"
     ;;
@@ -451,6 +457,8 @@ AS_IF([test x$with_pic = xyes], [X86ASMDEFS="${X86ASMDEFS} -DPIC"])
 AC_SUBST([X86ASMFLAGS])
 AC_SUBST([X86ASMDEFS])
 
+AM_CONDITIONAL([HAVE_EMSCRIPTEN], [test "${SYS}" = "emscripten"])
+
 dnl
 dnl Sadly autoconf does not think about testing foo.exe when ask to test
 dnl for program foo on win32
-- 
2.32.0

