From c3b56957dffe04521ed20e40ad5821c1b8148bc0 Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdi@videolabs.io>
Date: Sun, 30 May 2021 22:00:12 +0200
Subject: [PATCH 10/18] contrib: postproc: add support for wasm-emscripten

---
 contrib/src/postproc/rules.mak | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/contrib/src/postproc/rules.mak b/contrib/src/postproc/rules.mak
index 655acd5217..d9adb3cb32 100644
--- a/contrib/src/postproc/rules.mak
+++ b/contrib/src/postproc/rules.mak
@@ -106,6 +106,9 @@ endif
 ifdef HAVE_NACL
 POSTPROCCONF += --target-os=linux
 endif
+ifdef HAVE_EMSCRIPTEN
+POSTPROCCONF += --arch=wasm32 --target-os=linux
+endif
 
 # Build
 
-- 
2.32.0

