From 9b46de215341f5ad54f2215c4bc41fa8e40cc1e9 Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdisabwat@gmail.com>
Date: Tue, 1 Oct 2019 19:34:46 +0200
Subject: [PATCH 2/6] wip: adapt code to 4.0 api

---
 modules/audio_output/openal.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/modules/audio_output/openal.c b/modules/audio_output/openal.c
index 1c0b97a358..6d4a707642 100644
--- a/modules/audio_output/openal.c
+++ b/modules/audio_output/openal.c
@@ -84,7 +84,7 @@ static ALenum openalFormat = AL_FORMAT_STEREO_FLOAT32;
  * This structure is part of the audio output thread descriptor.
  * It describes the direct sound specific properties of an audio device.
  *****************************************************************************/
-struct aout_sys_t
+typedef struct aout_sys_t
 {
 	ALCdevice *dev;
 	ALCcontext *ctx;
@@ -94,7 +94,7 @@ struct aout_sys_t
 	ALuint source;
 	ALuint is_playing;
 	ALuint next_buffer;
-};
+} aout_sys_t;
 
 /*****************************************************************************
  * Module descriptor
@@ -269,6 +269,7 @@ static void Flush( audio_output_t *aout, bool wait )
 
 static int DeviceSelect( audio_output_t *aout, const char *psz_device)
 {
+  aout_sys_t *sys = aout->sys;
 	/*if ( psz_device == NULL )
 		return VLC_EGENERIC;
 	char* psz_end;
@@ -278,8 +279,9 @@ static int DeviceSelect( audio_output_t *aout, const char *psz_device)
 	if (aout->sys->dev == (ALCdevice*)ptr)
 		return VLC_SUCCESS;*/
 	const ALCchar* devInputName = alcGetString(NULL, ALC_DEFAULT_DEVICE_SPECIFIER); 
-	aout->sys->dev = alcOpenDevice(devInputName);
-	var_SetAddress (aout->obj.parent, "emscripten-openal-device", aout->sys->dev );
+
+	sys->dev = alcOpenDevice(devInputName);
+	var_SetAddress (vlc_object_parent(aout), "emscripten-openal-device", sys->dev );
 	aout_RestartRequest( aout, AOUT_RESTART_OUTPUT );
 	return VLC_SUCCESS;
 }
-- 
2.23.0

