From b904cb2019ff494a49d249b52f50e71fd4d1e83b Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdisabwat@gmail.com>
Date: Fri, 11 Oct 2019 11:47:48 +0200
Subject: [PATCH 6/6] fixup: implement timeget stub correctly

---
 modules/audio_output/openal.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/modules/audio_output/openal.c b/modules/audio_output/openal.c
index 374f2fd0a9..48fdadb9dd 100644
--- a/modules/audio_output/openal.c
+++ b/modules/audio_output/openal.c
@@ -55,7 +55,7 @@ static void Stop		( audio_output_t *);
 static void Play      	        ( audio_output_t *, block_t * , vlc_tick_t);
 static void Flush		( audio_output_t *, bool );
 static void Pause		( audio_output_t *, bool , vlc_tick_t);
-static int Time_Get		( audio_output_t *, mtime_t * );
+static int Time_Get		( audio_output_t *, vlc_tick_t * );
 static int Start		( audio_output_t *, audio_sample_format_t *restrict );
 static int DeviceSelect	( audio_output_t *, const char *);
 static int Volume_Set           ( audio_output_t *, float);
@@ -288,11 +288,9 @@ static int DeviceSelect( audio_output_t *aout, const char *psz_device)
 	return VLC_SUCCESS;
 }
 
-static int Time_Get( audio_output_t *aout, mtime_t *delay)
+static int Time_Get( audio_output_t *aout, vlc_tick_t *delay)
 {
-	aout_sys_t *sys = aout->sys;
-	*delay = aout_TimeGetDefault(aout, delay);
-	return 0;
+	return aout_TimeGetDefault(aout, delay);
 }
 
 /*****************************************************************************
-- 
2.23.0

